devolving_music/lib/song_scores.py

```diff
@@ -12,165 +13,174 @@
-
         # these prepeak, peak, and post posteak have each been sorted
         # accordingly with energy
-
         # that have been scored with current comparisons
-
         # get a moving average corresponding to quality score
 
         return mvg_avg
 
-    @staticmethod
-    def check_quality(
+    def check_quality(self,
             song_submissions_sorted: Iterable["SongSubmission"],
             remove: int):
 
         song_submissions_pruned = song_submissions_sorted
 
         # do (song_submissions_sorted[i].quality_score - mvg_avg[i])
-
         # for every song in score submission
-
         # return n indices with smallest values
-
         # these indices are to be removed
+
         return song_submissions_pruned
 
-    @staticmethod
-    def get_final_list(comparison_submissions: Iterable["SongComparison"],
+    def get_dict_from_keys(self, new_keys):
+        new_dict=dict()
+        new_dict={key: self.song_score_dict[key] for key in new_keys}
+        return new_dict
+    
+    def get_final_list(self):
 
+        scored_submissions = self.get_scores().copy()
+        
+        # remove all song_submissions with no information
+        info_list = SongScores.get_info_sort(scored_submissions)
+        for sub_info in info_list:
+            i_score = scored_submissions[sub_info].info_score
+            if(i_score == 0):
+                del scored_submissions[sub_info]
+            else:
+                break
+        # sort by postpeakyness
+        peaky_list = SongScores.get_peak_sort(scored_submissions)
+        peak_loc = int(0.7*len(peaky_list))
+        # Break peaky_sorted into two bins pre peak and post peak
+        pre_peak = self.get_dict_from_keys(peaky_list[:peak_loc])
+        post_peak = self.get_dict_from_keys(peaky_list[peak_loc:])
+        # energy_sorted
+        comeup = SongScores.get_energy_sort(pre_peak)
+        cooldown = SongScores.get_energy_sort(post_peak)
+        cooldown = cooldown[::-1]
+
+        # final_list is the song submission keys properly sorted
+        final_list = comeup + cooldown
 
-                       song_submissions: Iterable["SongSubmission"]):
+        return final_list
 
-        scored_submissions = SongSubmission.get_scores(
-            comparison_submissions, song_submissions)
+    def get_quality_list(self,
+            length_limit=300):
 
-        # remove all song_submissions with no information
+        final_quality_list = []
 
-        # sort by peakyness
+        # if len(final_list) is above length limit 
+        # remove_index=check_quality(final_list,len(energy_sorted)-lengthlimit)
+        # final_quality_list=remove(energy_sorted,remove_index)
 
-        # peaky_sorted_submissions=get_peak_sort(scored_submissions)
+        return final_quality_list
+
+    ###
+    # static methods can be moved into helper function
+    ###
+    @staticmethod
+    #sorts in ascending order
+    def get_info_sort(song_dict: "dict[int, ScoreSuite]"):
```

Comment:
Why not just make this non-static? The input could never logically be anything besides `self.song_score_dict`.

Reply:
this is not necessarily true there may be times when we want to info sort some subset of song_score_dict in a similar way that energy score does. I figured best to have all the sorts be static rather than some be static and some non-static